CMakeLists.txt

```diff
@@ -1,26 +1,66 @@
 cmake_minimum_required(VERSION 3.17)
 #cmake_policy(SET CMP0048 NEW)
 
-project(GHEX VERSION 0.2 LANGUAGES CXX)
+set(GHEX_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+list(APPEND CMAKE_MODULE_PATH "${GHEX_MODULE_PATH}")
+
+include(ghex_version)
+make_version(GHEX 0 3 0)
+
+project(GHEX VERSION ${GHEX_VERSION} LANGUAGES C CXX)
 
 # ---------------------------------------------------------------------
 # CMake setup, C++ version, build type, modules, etc
 # ---------------------------------------------------------------------
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_EXTENSIONS OFF)
 if(NOT CMAKE_BUILD_TYPE) # AND NOT CMAKE_CONFIGURATION_TYPES)
     set(CMAKE_BUILD_TYPE "Debug" CACHE STRING "Choose the type of build." FORCE)
     set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 endif()
-set(GHEX_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
-list(APPEND CMAKE_MODULE_PATH "${GHEX_MODULE_PATH}")
-
 
 # ---------------------------------------------------------------------
 # Define main library
 # ---------------------------------------------------------------------
 include(ghex_common)
 
+# ---------------------------------------------------------------------
+# GHEX options
+# ---------------------------------------------------------------------
+option(GHEX_GIT_SUBMODULE "Check submodules during build" OFF)
+option(GHEX_USE_BUNDLED_LIBS "Use bundled 3rd party libraries" OFF)
+set(GHEX_NO_RMA OFF CACHE BOOL "Disable in-node RMA completely")
+set(GHEX_USE_XPMEM OFF CACHE BOOL "Set to true to use xpmem shared memory")
+set(GHEX_USE_XPMEM_ACCESS_GUARD OFF CACHE BOOL "Use xpmem to synchronize rma access")
+mark_as_advanced(GHEX_USE_XPMEM_ACCESS_GUARD)
+set(GHEX_ENABLE_PARMETIS_BINDINGS OFF CACHE BOOL "Set to true to build with ParMETIS bindings")
+set(GHEX_ENABLE_ATLAS_BINDINGS OFF CACHE BOOL "Set to true to build with Atlas bindings")
+set(GHEX_BUILD_FORTRAN OFF CACHE BOOL "True if FORTRAN bindings shall be built")
+set(GHEX_BUILD_PYTHON_BINDINGS OFF CACHE BOOL "Set to true to build Python bindings")
+set(GHEX_WITH_TESTING OFF CACHE BOOL "True if tests shall be built")
+if (GHEX_WITH_TESTING)
+    enable_testing()
+    # Enable CDash dashboard testing integration
+    include(CTest)
+endif()
+
+
+# ---------------------------------------------------------------------
+# Dependencies
+# ---------------------------------------------------------------------
+include(ghex_external_dependencies)
+target_link_libraries(ghex_common INTERFACE MPI::MPI_CXX)
```

Comment:
Are we sending on the C++ interfaces to MPI? As far as I know they have been deprecated years ago...

Reply:
No, we use the C interface. This seems just to be the way to use the cmake module.